### PR TITLE
this commit adds the ability to check for commits by article id

### PR DESCRIPTION
**Pull request by rorybush**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -113,3 +113,43 @@ describe("GET /api/articles/:article_id", () => {
...
 });
+
+describe("GET /api/articles/:article_id/comments", () => {
+  test("should return the comments for the article ID provided", () => {
+    return request(app)
+      .get("/api/articles/1/comments")
+      .expect(200)
+      .then((res) => {
+        const { comments } = res.body;
+        expect(comments).toBeInstanceOf(Array);
```
Comment (Pipe-Smoking-Rabbit) [inline __tests__/app.test.js:124]: line 124 is not needed because if it's passing in a forEach loop bellow and has length of 11 then it's already covered that it must be an array - you could save a line of code here

File: __tests__/app.test.js [hunk 2]
```diff
@@ -113,3 +113,43 @@ describe("GET /api/articles/:article_id", () => {
...
+        expect(comments).toBeSortedBy("created_at");
+      });
+  });
+
+  test("returns 404 error if article id does not exist", () => {
+    return request(app)
+      .get("/api/articles/9999/comments")
+      .expect(404)
+      .then((res) => {
+        console.log(res.body);
```
Comment (Pipe-Smoking-Rabbit) [inline __tests__/app.test.js:151]: don't forget to take out old console.logs

File: __tests__/app.test.js [hunk 3]
```diff
@@ -113,3 +113,43 @@ describe("GET /api/articles/:article_id", () => {
...
+  test("returns the comments sorted by most recently posted", () => {
+    return request(app)
+      .get("/api/articles/1/comments")
+      .expect(200)
+      .then((res) => {
+        const { comments } = res.body;
+        expect(comments).toBeSortedBy("created_at");
+      });
+  });
+
```
Comment (Pipe-Smoking-Rabbit) [inline __tests__/app.test.js:145]: missing one test here that if you run this endpoint on an article that has `no comments` - it should return a `200` and give back an empty array, in this instance it's still a happy path to receive no comments as long as the `article_id` is valid

File: models/news-model.js [hunk 4]
```diff
@@ -33,3 +33,17 @@ exports.selectArticlesById = (article_id) => {
...
+
+exports.selectArticleCommentsById = (article_id) => {
+  const query = `SELECT comment_id, votes, created_at, author, body 
+  FROM comments 
+  WHERE article_id = $1
+  ORDER BY created_at ASC;`;
+
+  return db.query(query, [article_id]).then((result) => {
+    if (result.rowCount === 0) {
+      return Promise.reject({ status: 404, msg: "No Article Found." });
```
Comment (Pipe-Smoking-Rabbit) [inline models/news-model.js:45]: this `if (result.rowCount === 0)` will give you a false negative when you include a test for an `article_id` that has no comments - a way around this is to call your `selectArticleById` model function in you controller as well! Because the model function `selectArticleById` is already written to throw an error when it can't find any matching articles

**Review by Pipe-Smoking-Rabbit — CHANGES_REQUESTED** [review]
Very good code! However importantly it needs one more test for when you use an `article_id` that has `0` comments it should return `200` and an empty array - if you need any guidance on how to do this still then don't be shy to put in an nchelp in the morning

File: controllers/news-controller.js [hunk 5]
```diff
@@ -39,7 +39,10 @@ exports.getArticlesById = (req, res, next) => {
 
 exports.getArticleCommentsById = (req, res, next) => {
   const { article_id } = req.params;
-  selectArticleCommentsById(article_id)
+  return Promise.all([
+    selectArticlesById(article_id),
+    selectArticleCommentsById(article_id),
+  ])
```
Comment (robert112233) [inline controllers/news-controller.js:45]: This is a great way to differentiate between invalid article and valid article but with no comments, well done 👍

**Review by robert112233 — APPROVED** [review]
Merge good work 👍
